flocker/provision/_libcloud.py

```diff
@@ -121,3 +123,122 @@ def get_image(driver, image_name):
         return [s for s in driver.list_images() if s.name == image_name][0]
     except IndexError:
         raise ValueError("Unknown image.", image_name)
+
+
+# _node isn't immutable, since libcloud provides new instances
+# with updated data.
```

Comment:
I don't understand this comment. Can you expand a little and maybe move it to the class docstring below.